Tasks/Common/kubernetes-common-v2/image-metadata-helper.ts

```diff
@@ -0,0 +1,189 @@
+}
+
+export function getImageResourceUrlParameter(imageId: string, matchPattern: RegExp): string {
+    const imageMatch = imageId.match(matchPattern);
+    if (imageMatch && imageMatch.length >= 1) {
+        return imageMatch[1];
+    }
+
+    return "";
+}
+
+function getUserEmail(): string {
+    const build = "build";
+    const buildReason = "schedule";
+    const hostType = tl.getVariable("System.HostType").toLowerCase();
+    let userEmail: string = "";
+    if (hostType === build && tl.getVariable("Build.Reason").toLowerCase() !== buildReason) {
+        userEmail = tl.getVariable("Build.RequestedForEmail");
+    }
+    else {
+        userEmail = tl.getVariable("Release.RequestedForEmail");
+    }
+
+    return userEmail;
+}
+
+function getDescription(): string {
+    // Todo: Should we have a particular description with deployment details?
+    const release = "release";
+    const hostType = tl.getVariable("System.HostType").toLowerCase();
+    const description: string = hostType === release ? tl.getVariable("Release.ReleaseDescription") : "";
+    return description;
+}
+
+function getEnvironmentResourceAddress(): string {
+    const environmentResourceName = tl.getVariable("Environment.ResourceName");
+    const environmentResourceId = tl.getVariable("Environment.ResourceId");
+    if (!environmentResourceName && !environmentResourceId) {
+        return "";
+    }
+
+    return util.format("%s/%s", environmentResourceName, environmentResourceId);
+}
+
+function getPipelineUrl(deployment: any): string {
+    let pipelineUrl: string = "";
+    if (deployment && deployment.metadata && deployment.metadata.annotations) {
+        pipelineUrl = deployment.metadata.annotations["azure-pipelines/runuri"];
+    }
+
+    return pipelineUrl;
+}
+
+function getServerUrl(clusterInfo: any): string {
+    let serverUrl: string = "";
+    let serverUrlMatch = clusterInfo.match(matchPatternForServerUrl);
+    if (serverUrlMatch && serverUrlMatch.length >= 1) {
+        serverUrl = serverUrlMatch[1];
+    }
+
+    return serverUrl;
+}
+
+function getManifestFilePaths(): string[] {
+    let manifestFilePaths: string[] = [];
+    const commandArguments = tl.getInput("arguments", false);
+    const filePathMatch: string[] = commandArguments.split(matchPatternForFileArgument);
+    if (filePathMatch && filePathMatch.length >= 0) {
+        filePathMatch.forEach(manifestPath => {
+            if (!!manifestPath) {
+                manifestFilePaths.push(manifestPath.trim())
+            }
+        });
+    }
+
+    return manifestFilePaths;
+}
+
+function getPlatform(): string {
+    let platform: string = "Custom";
+    const connectionType = tl.getInput("connectionType", true);
+    if (connectionType === "Azure Resource Manager") {
+        platform = "AKS";
+    }
+
+    return platform;
+}
+
+export function IsJsonString(str) {
+    try {
+        JSON.parse(str);
+    } catch (e) {
+        return false;
+    }
+    return true;
+}
```

Comment:
Could you add some L0 tests for the util functions in this file?

Reply:
Will add